Date: Apr 21 10:18:42 2017

From: ba51588aebd95db152c157f39093afab0feaabf0 May 26 11:35:12 2017

Subject: [PATCH] RDK-17468, RDKB-8940: Open Source Updates

	Upgrade gnutls from 2.12.23 to 2.12.24 as PREFERRED_VERSION
        to address known high risk security vulnerabilities.

	rebased from original configure_madness.patch

	libtasn1 provides an .pc file in upstream,
	and a libgcrypt.pc is added in libgcrypt recipe,
	so use Requires.private for these two.

Source: GITHUB


Upstream-Status: Submitted [ https://github.com/dlespiau/poky/blob/master/meta/recipes-support/gnutls/gnutls/gnutls-openssl.patch ]

Signed-off-by: ahajjar <ahajjar@cyberpointllc.com>
---
Index: gnutls-2.12.11/lib/configure.ac
===================================================================
--- gnutls-2.12.11.orig/lib/configure.ac
+++ gnutls-2.12.11/lib/configure.ac
@@ -126,8 +126,8 @@ lgl_INIT
 AC_CHECK_FUNCS(getrusage,,)
 AC_LIB_HAVE_LINKFLAGS(pthread,, [#include <pthread.h>], [pthread_mutex_lock (0);])
 
-LIBGNUTLS_LIBS="-L${libdir} -lgnutls $LIBS"
-LIBGNUTLS_CFLAGS="-I${includedir}"
+LIBGNUTLS_LIBS="-lgnutls $LIBS"
+LIBGNUTLS_CFLAGS=""
 AC_SUBST(LIBGNUTLS_LIBS)
 AC_SUBST(LIBGNUTLS_CFLAGS)
 
Index: gnutls-2.12.11/lib/gnutls.pc.in
===================================================================
--- gnutls-2.12.11.orig/lib/gnutls.pc.in
+++ gnutls-2.12.11/lib/gnutls.pc.in
@@ -20,6 +20,4 @@ Description: Transport Security Layer im
 URL: http://www.gnu.org/software/gnutls/
 Version: @VERSION@
 Libs: -L${libdir} -lgnutls
-Libs.private: @LTLIBGCRYPT@ @LTLIBNETTLE@ @NETTLE_LIBS@ @GNUTLS_ZLIB_LIBS_PRIVATE@
-@GNUTLS_REQUIRES_PRIVATE@
 Cflags: -I${includedir}
Index: gnutls-2.12.11/libextra/configure.ac
===================================================================
--- gnutls-2.12.11.orig/libextra/configure.ac
+++ gnutls-2.12.11/libextra/configure.ac
@@ -41,8 +41,8 @@ LIBGNUTLS_EXTRA_HOOKS
 
 xgl_INIT
 
-LIBGNUTLS_EXTRA_LIBS="-L${libdir} -lgnutls-extra $LZO_LIBS $LIBGNUTLS_LIBS"
-LIBGNUTLS_EXTRA_CFLAGS="-I${includedir}"
+LIBGNUTLS_EXTRA_LIBS="-lgnutls-extra $LZO_LIBS $LIBGNUTLS_LIBS"
+LIBGNUTLS_EXTRA_CFLAGS=""
 AC_SUBST(LIBGNUTLS_EXTRA_LIBS)
 AC_SUBST(LIBGNUTLS_EXTRA_CFLAGS)
 
Index: gnutls-2.12.11/libextra/gnutls-extra.pc.in
===================================================================
--- gnutls-2.12.11.orig/libextra/gnutls-extra.pc.in
+++ gnutls-2.12.11/libextra/gnutls-extra.pc.in
@@ -22,5 +22,5 @@ URL: http://www.gnu.org/software/gnutls/
 Requires: gnutls
 Version: @VERSION@
 Libs: -L${libdir} -lgnutls-extra
-Libs.private: @LIBGNUTLS_EXTRA_LIBS@
+Libs.private: -llzo2
 Cflags: -I${includedir}
